azure_ai_search_rag.ipynb

```diff
@@ -0,0 +1,908 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Map: 100%|██████████| 10000/10000 [00:00<00:00, 13667.24 examples/s]\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Dataset({\n",
+       "    features: ['id', 'title', 'content', 'arxiv_id'],\n",
+       "    num_rows: 10000\n",
+       "})"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "data = data.map(lambda x: {\n",
+    "    \"id\": x[\"id\"],\n",
+    "    \"title\": x[\"title\"],\n",
+    "    \"content\": x[\"content\"],\n",
+    "    \"arxiv_id\": x[\"arxiv_id\"],\n",
+    "})\n",
+    "\n",
+    "data = data.remove_columns([\n",
+    "    \"prechunk_id\", \"postchunk_id\", \"references\"\n",
+    "])\n",
+    "\n",
+    "data\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We need to define an embedding model to create our embedding vectors for retrieval, for that we will be using Mistral AI's `mistral-embed`. There is some cost associated with this model, so be aware of that (costs for running this notebook are <$1).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from mistralai.client import MistralClient\n",
+    "import getpass  # console.mistral.ai/api-keys/\n",
+    "\n",
+    "mistral_api_key = os.getenv(\"MISTRAL_API_KEY\") or getpass.getpass(\"Enter your Mistral API key: \")\n",
+    "\n",
+    "mistral = MistralClient(api_key=mistral_api_key)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "embed_model = \"mistral-embed\"\n",
+    "\n",
+    "embeds = mistral.embeddings(\n",
+    "    model=embed_model, input=[\"this is a test\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can view the dimensionality of our returned embeddings, which we'll need soon when initializing our vector index:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1024"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dims = len(embeds.data[0].embedding)\n",
+    "dims"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Index Embeddings into Azure AI Search\n",
+    "\n",
+    "Now we create our vector DB to store our vectors. For this, we need to set up an [Azure AI Search service](https://portal.azure.com/#create/Microsoft.SearchV2).\n",
```

Comment:
Update https://portal.azure.com/#create/Microsoft.SearchV2 to https://portal.azure.com/#create/Microsoft.Search

Reply:
I still see the old link even though this is resolved.